pkg/ring/replication_strategy.go

```diff
@@ -4,28 +4,43 @@ import (
 	"fmt"
 )
 
-// replicationStrategy decides, given the set of ingesters eligible for a key,
+type ReplicationStrategy interface {
+	// Filter out unhealthy instances and checks if there're enough instances
+	// for an operation to succeed. Returns an error if there are not enough
+	// instances.
+	Filter(cfg Config, instances []IngesterDesc, op Operation) (healthy []IngesterDesc, maxFailures int, err error)
```

Comment:
Would it make sense move avoid using Config here? Filter needs two parameters from Config, not entire Config.

Reply:
We may. Would you pass `replicationFactor` and `heartbeatTimeout` to the function?

Reply:
Passing RF and heartbeat timeout sounds good to me. WDYT?

Reply:
Done